### PR TITLE
Small docs change to include timing reference

### PR DESCRIPTION
**Pull request by anisehsani**
Made some changes suggested by Ravi in the Notion doc.

### CODE REVIEW

File: composer/core/data_spec.py [hunk 1]
```diff
@@ -19,11 +19,13 @@ class DataSpec:
     Args:
         dataloader (DataLoader): The dataloader.
 
-        num_samples (int, optional). If specified, the total number of samples in the dataset, across all ranks.
-            If not specified, then ``len(dataloader.dataset)`` is used (if this property is available).
-            Otherwise, the dataset is assumed to be unsized.
+        num_samples (int, optional). The total number of samples in an epoch, across all ranks. This
+            field is used by the composer time tracking :class:`~comoser.core.time.Timer`. If not specified, then
```
Comment (hanlint) [inline composer/core/data_spec.py:23]: typo `comoser` -> `composer`

File: composer/core/data_spec.py [hunk 2]
```diff
@@ -19,11 +19,13 @@ class DataSpec:
...
+            ``len(dataloader.dataset)`` is used (if this property is available). Otherwise, the dataset is
+            assumed to be unsized.
 
-        num_tokens (int, optional): If specified, the total number of tokens in the dataset.
+        num_tokens (int, optional): The total number of tokens in an epoch. This field is used by the composer
+            time tracking :class:`~comoser.core.time.Timer`.
```
Comment (hanlint) [inline composer/core/data_spec.py:28]: typo `comoser`

**Review by hanlint — APPROVED** [review]
Can you fix, and also check the resulting built documentation, to make sure the links came out correctly?

File: composer/core/data_spec.py (hunk 1 above)
Comment (anisehsani) [reply composer/core/data_spec.py:23]: that's embarrassing lol

**Pull request merged**